app/code/Magento/Ui/view/base/web/js/modal/modal-component.js

```diff
@@ -361,7 +361,44 @@ define([
 
             if (this.valid) {
                 this.closeModal();
+            } else {
+                this._focusInvalid(this.elems());
+            }
+        },
+
+        /**
+         * Focus to first invalid element
+         *
+         * @param {HTMLElement} elems
+         * @param {boolean} firstRun
+         * @returns {boolean}
+         */
+        _focusInvalid: function (elems, firstRun = true) {
+            var self = this;
+
+            if (firstRun) {
+                window.focusElement = false;
```

Comment:
It is a bad approach to use global scope to check some state. You always can get current state directly from an element. Also, approach with `firstRun` variable looks like a hack